### ISSUE TITLE
Unsupported right operand type for in with __getitem__

### ISSUE THREAD

**Issue by nielsbuwen** [python/mypy#9306]
Hi, i may have found a false-positive

This is my code

```python
class Collection:
    def __getitem__(self, item: int) -> float:
        if 0 <= item < 10:
            return item / 10
        raise IndexError(item)


c = Collection()
print(0.1 in c)  # prints True
print(0.15 in c)  # prints False
print(iter(c))  # prints <iterator object at 0x...>
```

When running `mypy file.py` mypy reports three errors here:

```
file.py:9: error: Unsupported right operand type for in ("Collection")
file.py:10: error: Unsupported right operand type for in ("Collection")
file.py:11: error: No overload variant of "iter" matches argument type "Collection"
file.py:11: note: Possible overload variant:
file.py:11: note:     def [_T] iter(__iterable, Iterable[_T]) -> Iterator[_T]
file.py:11: note:     <2 more non-matching overloads not shown>
Found 3 errors in 1 file (checked 1 source file)
```

I expect mypy to not report any error here. In fact the code runs fine. It looks like mypy doesn't know that `Collection` is iterable.

- `python --version` -> Python 3.8.1
- `mypy --version` -> mypy 0.790+dev.c2c72da2d378c4f2121236a4e7ce65328bc34f4f

I found a similar - still open - issue #2118. I'm not sure whether that issue contains a solution. It seems like the issue wanted to fix a class in the standard library.

**Comment by hauntsaninja** [comment]
Seems more like a dupe of https://github.com/python/mypy/issues/2220

**Comment by gvanrossum** [comment]
Closing as duplicate.
